src/fetch.ts

```diff
@@ -27,24 +27,47 @@ export interface $Fetch {
   raw<T = any, R extends ResponseType = 'json'>(request: FetchRequest, opts?: FetchOptions<R>): Promise<FetchResponse<MappedType<R, T>>>
 }
 
-export function setHeader (options: FetchOptions, _key: string, value: string) {
-  const key = _key.toLowerCase()
+export function normalizeHeaders (options: FetchOptions): Pick<Headers, 'get' | 'set'> {
   options.headers = options.headers || {}
-  if ('set' in options.headers) {
-    ;(options.headers as Headers).set(key, value)
-  } else if (Array.isArray(options.headers)) {
-    const existingHeader = options.headers.find(([header]) => header.toLowerCase() === key)
-    if (existingHeader) {
-      existingHeader[1] = value
-    } else {
-      options.headers.push([key, value])
+
+  if ('set' in options.headers) { return options.headers as Pick<Headers, 'get' | 'set'> }
+
+  if (Array.isArray(options.headers)) {
+    const headers = options.headers
+    const findHeader = (key: string) => headers.find(([header]) => header.toLowerCase() === key)
+    return {
+      get: key => findHeader(key.toLowerCase())?.[1] || null,
+      set: (_key, value) => {
+        const key = _key.toLowerCase()
+        const existingHeader = findHeader(key)
+        if (existingHeader) {
+          existingHeader[1] = value
+        } else {
+          headers.push([key, value])
+        }
+      }
+    }
+  }
+
+  const headers = options.headers
+  const findHeaderKey = (key: string) => Object.keys(headers).find(header => header.toLowerCase() === key)
```

Comment:
Perf: We could normalizer either Array into Object or Object into array entities to merge logic and reduce bundle overhead. I would suggest using Object as normalize base since gives us direct lookup possibility